Lib/compileall.py

```diff
@@ -172,13 +172,12 @@ def compile_file(fullname, ddir=None, force=False, rx=None, quiet=0,
     if stripdir is not None:
         fullname_parts = fullname.split(os.path.sep)
         stripdir_parts = stripdir.split(os.path.sep)
-        ddir_parts = list(fullname_parts)
 
-        for spart, opart in zip(stripdir_parts, fullname_parts):
-            if spart == opart:
-                ddir_parts.remove(spart)
-
-        dfile = os.path.join(*ddir_parts)
+        if stripdir_parts != fullname_parts[:len(stripdir_parts)]:
+            if quiet < 2:
+                print("The stripdir path {!r} is not a valid prefix for source path {!r}; ignoring".format(stripdir, fullname))
```

Comment:
Have you considered using a warning instead of print? I see some benefits in a warning over plain print.

Reply:
I wasn't sure if this question was answered by me and then I went away on vacation so I ended up forgetting about it entirely.

I considered using a warning at first, but since all other error indications are done with `print()` controlled using the `quiet` variable I decided it would be best to just stick to the way the compileall module was already sending output to the user.

Reply:
I'm fine with that and you are right that it matches the rest of the codebase.